runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -174,24 +183,29 @@ DataflowPipelineJob run(Pipeline pipeline, DataflowRunner runner) {
     return runner.apply(transform, input);
   }
 
-  Optional<Boolean> checkForSuccess(DataflowPipelineJob job)
+  /**
+   * Check success status of a dataflow job with given job metrics.
+   *
+   * @return Optional.of(false) if job failed/cancelled or PAssert failed,
+   *         Optional.of(true) if number of success PAssert meet expects,
+   *         or Optional.absent() otherwise.
```

Comment:
what is otherwise?
Return Optional.absent() if:
1. metrics is not available
2. PAssert conditions haven't failed or all passed.

Also, add:
<p>If no PAssert is used, it returns either Optional.of(true) or Optional.absent().